dapp/src/components/buySell/ApproveButtonLogic.js

```diff
@@ -0,0 +1,188 @@
+import { ethers } from 'ethers'
+import { fbt } from 'fbt-runtime'
+import { useStoreState } from 'pullstate'
+import React, {
+  forwardRef,
+  useEffect,
+  useImperativeHandle,
+  useState,
+} from 'react'
+import ContractStore from 'stores/ContractStore'
+import analytics from 'utils/analytics'
+import { assetRootPath } from 'utils/image'
+import { useWeb3React } from '@web3-react/core'
+import Media from 'react-media'
+
+// const GLOBAL_MEDIA_QUERIES = {
+//   small: "(max-width: 599px)",
+//   medium: "(min-width: 600px) and (max-width: 1199px)",
+//   large: "(min-width: 1200px)"
+// };
+// const matches = useMedia({ queries: GLOBAL_MEDIA_QUERIES });
+
+const ApproveButtonLogic = forwardRef(
+  (
+    {
+      formHasErrors,
+      swappingGloballyDisabled,
+      needsApproval,
+      allowButtonState,
+      setAllowButtonState,
+      onBuyNow,
+      storeTransaction,
+      storeTransactionError,
+      onMintingError,
+      rpcProvider,
+      swapMetadata,
+      swapMode,
+      coin,
+    },
+    ref
+  ) => {
+    const [contract, setContract] = useState(null)
+    const { library, account } = useWeb3React()
+
+    const {
+      vault,
+      flipper,
+      uniV3SwapRouter,
+      uniV2Router,
+      sushiRouter,
+      curveOUSDMetaPool,
+      usdt,
+      dai,
+      usdc,
+      ousd,
+    } = useStoreState(ContractStore, (s) => s.contracts || {})
+
+    const contractMap = {
```

Comment:
Names of the contract variables can be changed already deconstructing them from `ContractStore` like so:

```
const {
      vault,
      flipper,
      uniV3SwapRouter: uniswap,
      uniV2Router: uniswapV2,
      sushiRouter: sushiswap,
      curveOUSDMetaPool: curve,
      usdt,
      dai,
      usdc,
      ousd,
    } = useStoreState(ContractStore, (s) => s.contracts || {})
```

without the need for a `contractMap`

Reply:
I used the same way it was used before on the approve modal for a matter of consistence

Reply:
I see tnx, seems we could improve code there :)